jax/_src/numpy/lax_numpy.py

```diff
@@ -3334,6 +3334,207 @@ class _Ogrid(_IndexGrid):
 ogrid = _Ogrid()
 
 
+def _make_1d_grid_from_slice(s: slice):
+  start = s.start or 0
+  stop = s.stop
+  step = s.step or 1
+  if np.iscomplex(step):
+    newobj = linspace(start, stop, int(_abs(step)))
+  else:
+    newobj = arange(start, stop, step)
+
+  return newobj
+
+
+class _AxisConcat:
+  """Concatenates slices, scalars and array-like objects along a given axis."""
+  def __getitem__(self, key):
+    if not isinstance(key, tuple):
+      key = (key,)
+
+    params = [self.axis, self.ndmin, self.trans1d, -1]
+
+    if isinstance(key[0], str):
+      directive = key[0]
+      # check two special cases: matrix directives
+      mats = ("r", "c")
+      if directive in mats:
+        params[-1] = mats.index(directive)
```

Comment:
Slight preference here for the more direct
```
if directive == "r":
  params[-1] = 0
elif directive == "c":
  params[-1] = 1
else:
  ...
```

Reply:
Alright! Before I push this, do you think that the 

```
      # cut out the directive
      key = key[1:]
```
part at the very end of that if-else branch should move up top (just below the `directive` assignment) to improve legibility/flow?

Reply:
Sure, that's a good idea. You could even use something like this to be more concise:
```
directive, *key = key
```